src/libs/actions/IOU.js

```diff
@@ -779,110 +772,213 @@ function splitBillAndOpenReport(participants, currentUserLogin, currentUserAccou
 }
 
 /**
- * @param {String} chatReportID
- * @param {String} iouReportID
- * @param {Object} moneyRequestAction - the money request reportAction we are deleting
- * @param {Boolean} shouldCloseOnDelete
+ * @param {String} transactionID
+ * @param {Object} reportAction - the money request reportAction we are deleting
+ * @param {Boolean} isSingleTransactionView
  */
-function deleteMoneyRequest(chatReportID, iouReportID, moneyRequestAction, shouldCloseOnDelete) {
-    const iouReport = iouReports[`${ONYXKEYS.COLLECTION.REPORT}${iouReportID}`];
-    const transactionID = moneyRequestAction.originalMessage.IOUTransactionID;
+function deleteMoneyRequest(transactionID, reportAction, isSingleTransactionView = false) {
+    // STEP 1: Get all collections we're updating
+    const iouReport = allReports[`${ONYXKEYS.COLLECTION.REPORT}${reportAction.originalMessage.IOUReportID}`];
+    const chatReport = allReports[`${ONYXKEYS.COLLECTION.REPORT}${iouReport.chatReportID}`];
+    const reportPreviewAction = ReportActionsUtils.getReportPreviewAction(iouReport.chatReportID, iouReport.reportID);
+    const transaction = allTransactions[`${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`];
+    const transactionThreadID = reportAction.childReportID;
+    let transactionThread = null;
+    if (transactionThreadID) {
+        transactionThread = allReports[`${ONYXKEYS.COLLECTION.REPORT}${transactionThreadID}`];
+    }
 
-    // Get the amount we are deleting
-    const amount = moneyRequestAction.originalMessage.amount;
-    const optimisticIOUAction = ReportUtils.buildOptimisticIOUReportAction(
-        CONST.IOU.REPORT_ACTION_TYPE.DELETE,
-        amount,
-        moneyRequestAction.originalMessage.currency,
-        moneyRequestAction.originalMessage.comment,
-        [],
-        transactionID,
-        '',
-        iouReportID,
-    );
+    // STEP 2: Decide if we need to:
+    // 1. Delete the transactionThread - delete if there are no visible comments in the thread
+    // 2. Update the iouPreview to show [Deleted request] - update if the transactionThread exists AND it isn't being deleted
+    const shouldDeleteTransactionThread = transactionThreadID ? ReportActionsUtils.getLastVisibleMessage(transactionThreadID).lastMessageText.length === 0 : false;
```

Comment:
We missed to take in to account that there could be `MODIFIEDEXPENSE` actions which would make the last visible message as non-empty, and thus, prevent the thread from getting deleted.